File: mhd1d/ideal_gas.py
"""
Relations for ideal gas
"""

import numpy as np

gamma = 5/3

def ideal_gas_law(rho, e):
    """
    Equation of state for ideal gas
    
    Parameters
    ----------
    rho : float or array
        mass density
    e : float or array
        energy density
    
    Returns
    -------
    P_g : float or array
        gas pressure
    """

    return rho*(gamma - 1)*e

def sound_speed(rho, P):
    """
    Sound speed in an ideal gas

    Parameters
    ----------
    rho : float or array
        mass density
    P_g : float or array
        gas pressure
    
    Returns
    -------
    c_s : float or array
        sound speed
    """

    return np.sqrt(gamma*P/rho)

File: julia/ideal_gas.jl
import PhysicalConstants.CODATA2018: m_u, k_B

using Unitful

const gamma = 5/3
# Average particle mass
const μ = 1.0

"""
    ideal_gas(value::AbstractFloat, mode=:pressure::Symbol)

Equation of state for ideal gas
"""
function ideal_gas(x::AbstractFloat; mode::Symbol=:pressure)

    if mode==:pressure
        # input is internal energy, calculates the gas pressure
        P = (gamma - 1)*x
        return P

    elseif mode==:energy
        # input is pressure, calculates the internal energy
        e = x/(gamma - 1)
        return e
    
    else
        throw(ErrorException("Mode $mode does not exist"))
    end

end

"""
    ideal_gas(e::AbstractFloat, rho::AbstractFloat)

Calculates temperature in an ideal gas
"""
function ideal_gas(e::AbstractFloat, rho::AbstractFloat)

    (gamma - 1)*e/rho*μ*ustrip(m_u/k_B)

end

"""
    sound_speed(P::AbstractFloat, rho::AbstractFloat)

Sound speed in an ideal gas
"""
function sound_speed(P::AbstractFloat, rho::AbstractFloat)

    sqrt(gamma*P/rho)

end